lib/logwriter.h

```diff
@@ -74,7 +77,44 @@ typedef struct _LogWriterOptions
   gint mark_freq;
 } LogWriterOptions;
 
-typedef struct _LogWriter LogWriter;
+typedef struct _LogWriter
+{
+  LogPipe super;
+  LogQueue *queue;
+  guint32 flags:31;
+  gint32 seq_num;
+  StatsCounterItem *dropped_messages;
+  StatsCounterItem *suppressed_messages;
+  StatsCounterItem *processed_messages;
+  StatsCounterItem *stored_messages;
+  LogPipe *control;
+  LogWriterOptions *options;
+  LogMessage *last_msg;
+  guint32 last_msg_count;
+  GString *line_buffer;
+
+  gint stats_level;
+  guint16 stats_source;
+  gchar *stats_id;
+  gchar *stats_instance;
+
+  struct iv_fd fd_watch;
+  struct iv_timer suspend_timer;
+  struct iv_task immed_io_task;
+  struct iv_event queue_filled;
+  MainLoopIOWorkerJob io_job;
+  GStaticMutex suppress_lock;
+  MlBatchedTimer suppress_timer;
+  MlBatchedTimer mark_timer;
+  struct iv_timer reopen_timer;
+  gboolean work_result;
+  gint pollable_state;
+  LogProtoClient *proto, *pending_proto;
+  gboolean watches_running:1, suspended:1, working:1, waiting_for_throttle:1;
+  gboolean pending_proto_present;
+  GCond *pending_proto_cond;
+  GStaticMutex pending_proto_lock;
+} LogWriter;
 
```

Comment:
eh, this one is bad. I wouldn't want to publish the entire LogWriter internals (especially is it will pull in further dependencies in the interface which should belong to the implementation).

maybe we should keep the cast in this case? that's bad, but still better than publishing LogWriter this way.